src/main/java/io/confluent/connect/hdfs/wal/WALFile.java

```diff
@@ -194,40 +197,52 @@ public static Option blockSize(long value) {
                                            "compatible with stream");
       }
 
+      FileSystem fs = null;
       FSDataOutputStream out;
       boolean ownStream = fileOption != null;
-      if (ownStream) {
-        Path p = fileOption.getValue();
-        FileSystem fs;
-        fs = p.getFileSystem(conf);
-        int bufferSize = bufferSizeOption == null ? getBufferSize(conf) :
-                         bufferSizeOption.getValue();
-        short replication = replicationOption == null ?
-                            fs.getDefaultReplication(p) :
-                            (short) replicationOption.getValue();
-        long blockSize = blockSizeOption == null ? fs.getDefaultBlockSize(p) :
-                         blockSizeOption.getValue();
-
-        if (appendIfExistsOption != null && appendIfExistsOption.getValue()
+
+      try {
+        if (ownStream) {
+          Path p = fileOption.getValue();
+          fs = p.getFileSystem(conf);
+          int bufferSize = bufferSizeOption == null ? getBufferSize(conf) :
+            bufferSizeOption.getValue();
+          short replication = replicationOption == null ?
+            fs.getDefaultReplication(p) :
+            (short) replicationOption.getValue();
+          long blockSize = blockSizeOption == null ? fs.getDefaultBlockSize(p) :
+            blockSizeOption.getValue();
+
+          if (appendIfExistsOption != null && appendIfExistsOption.getValue()
             && fs.exists(p)) {
-          // Read the file and verify header details
-          try (WALFile.Reader reader =
+            // Read the file and verify header details
+            try (WALFile.Reader reader =
                    new WALFile.Reader(conf, WALFile.Reader.file(p), new Reader.OnlyHeaderOption())){
-            if (reader.getVersion() != VERSION[3]) {
-              throw new VersionMismatchException(VERSION[3], reader.getVersion());
+              if (reader.getVersion() != VERSION[3]) {
+                throw new VersionMismatchException(VERSION[3], reader.getVersion());
+              }
+              sync = reader.getSync();
             }
-            sync = reader.getSync();
+            out = fs.append(p, bufferSize);
+            this.appendMode = true;
+          } else {
+            out = fs.create(p, true, bufferSize, replication, blockSize);
           }
-          out = fs.append(p, bufferSize);
-          this.appendMode = true;
         } else {
-          out = fs.create(p, true, bufferSize, replication, blockSize);
+          out = streamOption.getValue();
         }
-      } else {
-        out = streamOption.getValue();
+
+        init(conf, out, ownStream);
+      } catch (RemoteException re) {
+        log.error("Failed creating a WAL Writer: " + re.getMessage());
```

Comment:
Yes, I was hoping to log a bit more information so that it is easier to read the debug log which is VERY long. 

So re.getMessage() is the short version that I like :-) If we log the stack trace, the same info will be logged again all the way up at https://github.com/confluentinc/kafka-connect-hdfs/blob/master/src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java#L324.